### ISSUE TITLE
how to increase sgx.thread_num in in the manifest correctly?

### ISSUE THREAD

**Issue by MatthewShine** [oscarlab/graphene#699]
When i try to run a python pgm using numpy lib ,the msg bellow come out.
```
There are no available TCS pages left for a new thread!
Please try to increase sgx.thread_num in the manifest.
The current value is 4
```
fellow the msg ,I edit the  python.manifest.sgx 
```
sgx.enclave_size = 256M
sgx.isvprodid = 0
sgx.isvsvn = 0
sgx.sigfile = file:python.sig
sgx.static_address = 1
sgx.thread_num = 16
```
once i changed the parm inside the python.manifest.sgx,Here is the err msg (no more info )
```
initializing enclave failed: init_enclave: 1

```

I think there should be some rules about incresing the thread_num. Does anyone can tell me how to solve it?

**Comment by dimakuv** [comment]
Ok, there are several things going on here.

1. You shouldn't modify `*.manifest.sgx` file. This file is autogenerated during `make SGX=1` (or if you know how to build out-of-Graphene-tree, while running the `pal-sgx-sign` script). You should modify the template file which is usually named `*.manifest.template` (in some cases simply `*.manifest`).

2. The autogenerated `python.manifest.sgx` is included in the MRENCLAVE measurement of the Python enclave (in the sense that the contents of this file are thrown into the measurement mix). So, every time you run `make SGX=1` (which generates this file), you also need to run `make SGX_RUN=1` (or if you know how to build out-of-Graphene-tree, run the `pal-sgx-get-token` script). This will update the SIGSTRUCT of your enclave which is consulted during the initialization phase.

Basically, you need to do three things:
1. Add `sgx.thread_num = 16` to `python.manifest.template`.
2. Rebuild your Python with `make SGX=1` (autogenerates `python.manifest.sgx`).
3. Acquire a new SIGSTRUCT with `make SGX_RUN=1`.

Now you can run Python and hopefully not get error messages.

**Comment by MatthewShine** [comment]
Thanks for replying. It does works.
